utils/src/bitwise_operations.rs

```diff
@@ -27,6 +37,28 @@ impl BitwiseOps for BigUint {
         )
     }
 
+    fn bitwise_not(input: &BigUint, bits: Option<usize>) -> BigUint {
+        // pad if needed / desired
+        // first get the number of bits of the input,
+        // take into account that BigUint::bits() returns 0 if the input is 0
+        let in_bits = input.bitlen() as usize;
+        let bits = max(in_bits, bits.unwrap_or(0));
+        // build vector of bits in little endian (least significant bit in position 0)
+        let mut bit_vec = vec![];
+        // negate each of the bits of the input
+        (0..bits).for_each(|i| bit_vec.push(!Self::bit_at(input, i as u32)));
+        le_bitvec_to_biguint(&bit_vec)
+    }
+
+    // Returns the bit value of a BigUint input at a certain position or zero
```

Comment:
I'm not sure this interface is sufficient.  You cannot discern an invalid position.  And it doesn't check the index at the start either.

Reply:
Mmmm I am not sure I follow. This check is indeed performed because the `.bits()` function returns whether a bit at a certain position is set or not. Meaning, I don't do it manually, but the `bits` does it for us.

Reply:
I mean this.  If I call `bit_at(x, 23939)` it will return `false` which means the bit at position `23939` is not set.  However, it should instead fail so that the caller knows it's a bad index.

Reply:
`bit_at` could be part of `BigUintBitwiseHelpers` trait so you could just to `big.bit_at(x)`

Reply:
Oh, but I want it to be zero when the index is not set

Reply:
Because I use this when I perform bitwise operation, and it may be the case that two inputs have different bit length, but that shouldn't prevent me from performing these operations. Instead, I want it to take zero when it is not set.

Reply:
In any case I am only using `bit_at` internally as an auxiliary function for the bitwise operations. Out of this scope perhaps your interpretation could make more sense, but I explicitly need it to have this behaviour within bitwise operations module.